src/hooks/useAccessGrantRemovalDescriptions.ts

```diff
@@ -0,0 +1,96 @@
+import { useUserStore } from 'context/User/useUserContextStore';
+import { useCallback, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import {
+    useEntitiesStore_capabilities_adminable,
+    useEntitiesStore_capabilities_readable,
+} from 'stores/Entities/hooks';
+import { BaseGrant, Grant_UserExt } from 'types';
+import { ESTUARY_SUPPORT_ROLE, isGrant_UserExt } from 'utils/misc-utils';
+import { useShallow } from 'zustand/react/shallow';
+
+type GrantScopeMessageIdSuffix =
+    | 'finalEmail'
+    | 'email'
+    | 'ownEmail'
+    | 'ownTenant'
+    | 'support'
+    | 'tenant';
+
+export type AccessGrantRemovalType = 'dangerous' | 'normal';
```

Comment:
This feels like more of a _severity_ than a _type_ (i.e., `AccessGrantRemovalSeverity`).

_Thinking out loud._ I could see us needing an action severity type that identifies the relative impact of an action (i.e., `'critical'` or `'high'`, `'moderate'` or `'medium'`, etc.).

Reply:
I like severity